pkg/vadmin/create_db_vc.go

```diff
@@ -17,14 +17,71 @@ package vadmin
 
 import (
 	"context"
-	"fmt"
+	"strings"
 
+	vops "github.com/vertica/vcluster/vclusterops"
+	vapi "github.com/vertica/vertica-kubernetes/api/v1beta1"
+	"github.com/vertica/vertica-kubernetes/pkg/net"
 	"github.com/vertica/vertica-kubernetes/pkg/vadmin/opts/createdb"
 	ctrl "sigs.k8s.io/controller-runtime"
 )
 
 // CreateDB will construct a new DB using the vcluster-ops library
 func (v VClusterOps) CreateDB(ctx context.Context, opts ...createdb.Option) (ctrl.Result, error) {
 	v.Log.Info("Starting vcluster CreateDB")
-	return ctrl.Result{}, fmt.Errorf("not implemented")
+
+	// get the certs
+	certs, err := v.retrieveHTTPSCerts(ctx)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// get create_db options
+	s := createdb.Parms{}
+	s.Make(opts...)
+
+	// call vcluster-ops library to create db
+	vopts := v.genCreateDBOptions(&s, certs)
+	vdb, err := v.VCreateDatabase(&vopts)
+	if err != nil {
+		v.Log.Error(err, "failed to create a database")
+		return ctrl.Result{}, err
+	}
+
+	v.Log.Info("Successfully created a database", "dbName", vdb.Name)
+	return ctrl.Result{}, nil
+}
+
+func (v VClusterOps) genCreateDBOptions(s *createdb.Parms, certs *HTTPSCerts) vops.VCreateDatabaseOptions {
+	opts := vops.VCreateDatabaseOptionsFactory()
+
+	opts.RawHosts = s.Hosts
+	v.Log.Info("Create database on the hosts: " + strings.Join(s.Hosts, ","))
```

Comment:
minor: can we pass the hosts as separate parms?
```suggestion
	v.Log.Info("Setup create db options", "hosts", strings.Join(s.Hosts, ","))
```